site/_posts/2022-09-09-what-is-webauthn-why-do-you-care.md

```diff
@@ -0,0 +1,193 @@
+
+An authenticator is simply a piece of hardware or software that fulfills these criteria:
+
+* It knows how to talk CTAP2
+* It can safely generate public and private keys
+
+That's it.
+
+There are two major categories of authenticators: platform and cross-platform. Platform authenticators are tied to one device, like a computer or tablet. They are often implemented in software, and can be built into the operating system as well.
+
+Cross-platform authenticators are independent pieces of hardware which can be plugged into different devices. Examples of cross-platform authenticators include Yubikeys or Trezors.
+
+There are options when it comes to authenticators, but what makes WebAuthn really exciting is that all of the major operating systems have implemented platform authenticators.
+
+The operating systems often offer authentication methods which would be difficult for normal web developers to build or access outside of WebAuthn. This includes:
+
+* Face ID from Apple
+* Android fingerprint identification
+* Windows Hello facial recognition
+
+If you implement WebAuthn, you gain access to all of these authentication methods with little effort.
+
+There are other authenticators which serve different purposes. The hardware authenticators fit certain use cases such as workforce or employee login. But they aren't a fit for for the Cosmo's Clown Store consumer ecommerce use case; normal folks typically don't have hardware authenticators.
+
+## What does a WebAuthn process look like?
+
+Let's get down to the nuts and bolts of implementing WebAuthn. To authenticate a user, you have to call this method in the browser. (It is a method on the `navigator` object because we're never going to forget Netscape Navigator.)
+
+```javascript
+navigator.credentials.get(options);
+```
+
+Of course, you might be wondering what is in the `options` object. Here's an excerpt, but there are quite a few variations. 
+
+```javascript
+options =  {
+  publicKey: {
+    rpId: "cosmosclownstore.com", // the identifier of the website
+    challenge: “..." // one-time random string to prevent replay attacks
+    userVerification: "preferred", // if you want the user to authenticate or just indicate presence
+    // and more!
+  }
+}
+```
+
+When you make this function call, a request is made to the authenticator which interacts with the user. This is the part that CTAP2 takes care of. If all goes well, the user proves their identity and the authenticator populates and returns a results object, the `assertionResponse` below.
+
+```javascript
+navigator.credentials.get(options)
+  .then(function (assertionResponse) {
+    // Send authentication status to server 
+  }).catch(function (err) {
+    // No acceptable authenticator or user refused. 
+  });
+```
+
+The following is a subset of the `assertionResponse` object, which has a number of fields:
+
+```javascript
+{
+  signature : ..., // the signature generated by the private key over the contents of the response
```

Comment:
Maybe worth mentioning that the challenge is part of the signed data? Or perhaps it's not worth getting into that level of detail here?

Reply:
Thanks, this is not a reference doc, so I think that's too much detail.